packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -566,6 +576,15 @@ export abstract class FunctionBase extends Resource implements IFunction, ec2.IC
       return (principal as iam.ArnPrincipal).arn;
     }
 
+    if (principal instanceof iam.OrganizationPrincipal) {
```

Comment:
are these `instanceof`'s okay? Reading the comment on line 551/561, "after v2, we can change these to `instanceof`" which makes me think that new additions can use `instanceof` directly.

Reply:
It's actually not true. That comment is a lie.

Reply:
removed the "lie" then, and used duck-typing